### PR TITLE
Add privateLinkConnectionProperties to network interface.

### PR DESCRIPTION
**Pull request by bowen0106**
Adding one property in network interface to display private link connection properties.

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by anton-evseev — CHANGES_REQUESTED** [review]
Need to target 2019-07-01

File: specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkInterface.json [hunk 1]
```diff
@@ -897,10 +897,34 @@
...
     },
+    "NetworkInterfaceIPConfigurationPrivateLinkConnectionProperties": {
+      "properties": {
+        "groupId": {
+          "type": "string",
+          "description": "The group id for current private link connection."
+        },
+        "requiredMemberName": {
+          "type": "string",
+          "description": "The require member name for current private link connection."
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkInterface.json]:
```suggestion
          "description": "The required member name for current private link connection."
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkInterface.json [hunk 2]
```diff
@@ -897,10 +897,34 @@
...
+          "description": "PrivateLinkConnection properties for the network interface."
         }
       },
       "description": "Properties of IP configuration."
     },
+    "NetworkInterfaceIPConfigurationPrivateLinkConnectionProperties": {
+      "properties": {
+        "groupId": {
+          "type": "string",
+          "description": "The group id for current private link connection."
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkInterface.json]:
```suggestion
          "description": "The group ID for current private link connection."
```

**Review by anton-evseev — APPROVED** [review]
LGTM except a few nitpicks about descriptions

**Comment by number213** [comment]
Mirrored change from https://github.com/Azure/azure-rest-api-specs/pull/7179 to fix CI errors. Can't really cherry pick that commit since ADO configs were refactored in master branch. Before merging network branch back into master I'll rebase it and remove my commit

**Comment by number213** [comment]
@pilor @KrisBash please review this from ARM's side

File: specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkInterface.json [hunk 3]
```diff
@@ -897,10 +897,34 @@
         "provisioningState": {
           "$ref": "./network.json#/definitions/ProvisioningState",
           "description": "The provisioning state of the network interface IP configuration."
+        },
+        "privateLinkConnectionProperties": {
```
Comment (pilor) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkInterface.json]: The "properties" suffix seems unnecessary.  "publicIPAddress" and "subnet" aren't using the suffix and its not typical across other resources.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkInterface.json [hunk 4]
```diff
@@ -897,10 +897,34 @@
...
+          "$ref": "#/definitions/NetworkInterfaceIPConfigurationPrivateLinkConnectionProperties",
+          "description": "PrivateLinkConnection properties for the network interface."
         }
       },
       "description": "Properties of IP configuration."
     },
+    "NetworkInterfaceIPConfigurationPrivateLinkConnectionProperties": {
```
Comment (pilor) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkInterface.json]: Addition of a new property should involve a new api-version.  See https://microsoft.sharepoint.com/:w:/t/azureresourcemanagerteam/EWXsAQ1yx25KkyYCeeWGUgwBSBxEdUDEbHi6FZ__U8EOQw?e=xEcppo  (specifically the "New property added to response" example).

**Review by pilor — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkInterface.json (hunk 4 above)
Comment (number213) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkInterface.json]: This is a new API version, `2019-07-01`, isn't not merged into master yet, and there are no SDKs published that use it

File: specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkInterface.json (hunk 3 above)
Comment (bowen0106) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkInterface.json]: Hi Pilor, 
So regarding to your question,  the property's name is '**privateLinkConnectionProperties**' from the Networking RP server side, it's been public and we cannot change it. So I think we should keep the name as is. Thank you.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkInterface.json (hunk 4 above)
Comment (pilor) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkInterface.json]: ah, didn't realize this was into a feature branch

File: specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkInterface.json (hunk 3 above)
Comment (pilor) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkInterface.json]: Public how?  If this is already public then what am I reviewing here?

Comment (bowen0106) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkInterface.json]: Hi Pilor,
My meaning is the NRP side has already released the REST api for this part. And the property is already there for a while which named 'privateLinkConnectionProperties'. So I don't think there is a reason to change the name.
I attached one of example of return json string for network interface for your reference:
```
{
  "name": "PEVnet1.nic.b9414b57-e682-40b9-ada8-312e40e2c88b",
  "id":
"/subscriptions/15aa9ba4-9bd3-499f-ba12-9a5736d0cd68/resourceGroups/PrivateEndpointRunnerTestToStaticRGRGWestcentralusSLBV2Dev/providers/Microsoft.Network/networkInterfaces/PEVnet1.nic.b9414b57-e682-40b9-ada8-312e40e2c88b",
  "etag": "W/\"b2dd33b7-92d1-460a-baa6-d5f55960e4aa\"",
  "properties": {
    "provisioningState": "Succeeded",
    "resourceGuid": "7e3a996f-09a0-41e2-9f6d-0841110f7823",
    "ipConfigurations": [
      {
        "name": "sqlServer-sqlServer.privateEndpoint",
        "id":
"/subscriptions/15aa9ba4-9bd3-499f-ba12-9a5736d0cd68/resourceGroups/PrivateEndpointRunnerTestToStaticRGRGWestcentralusSLBV2Dev/providers/Microsoft.Network/networkInterfaces/PEVnet1.nic.b9414b57-e682-40b9-ada8-312e40e2c88b/ipC
onfigurations/sqlServer-sqlServer.privateEndpoint",
        "etag": "W/\"b2dd33b7-92d1-460a-baa6-d5f55960e4aa\"",
        "type": "Microsoft.Network/networkInterfaces/ipConfigurations",
        "properties": {
          "provisioningState": "Succeeded",
          "privateIPAddress": "10.0.0.4",
          "privateIPAllocationMethod": "Dynamic",
          "subnet": {
            "id":
"/subscriptions/15aa9ba4-9bd3-499f-ba12-9a5736d0cd68/resourceGroups/PrivateEndpointRunnerTestToStaticRGRGWestcentralusSLBV2Dev/providers/Microsoft.Network/virtualNetworks/vNet1-nr3kriysgtcxk/subnets/subnet1"
          },
          "primary": true,
          "privateIPAddressVersion": "IPv4",
          "privateLinkConnectionProperties": {
            "groupId": "sqlServer",
            "requiredMemberName": "sqlServer",
            "fqdns": [
              "sql-for-test-4nuppzq4x37iq.database.windows.net"
            ]
          }
        }
      }
    ],
    "dnsSettings": {
      "dnsServers": [],
      "appliedDnsServers": [],
      "internalDomainNameSuffix": "vji40krafe4ujcvrvvzyhdrltc.yx.internal.cloudapp.net"
    },
    "macAddress": "",
    "enableAcceleratedNetworking": false,
    "enableIPForwarding": false,
    "hostedWorkloads": [],
    "tapConfigurations": [],
    "privateEndpoint": {
      "id": "/subscriptions/15aa9ba4-9bd3-499f-ba12-9a5736d0cd68/resourceGroups/PrivateEndpointRunnerTestToStaticRGRGWestcentralusSLBV2Dev/providers/Microsoft.Network/privateEndpoints/PEVnet1"
    }
  },
  "type": "Microsoft.Network/networkInterfaces"
}
```

**Review by KrisBash — APPROVED** [review]
signing off for ARM

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**